.github/workflows/docs.yml

```diff
@@ -0,0 +1,50 @@
+name: Build and deploy docs
+
+on:
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Install dependencies
+        working-directory: ./docs
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-docs.txt
```

Comment:
this step should use `pip install .'[doc]'` since the build deps are specified in `setup.cfg` and not in `docs/requirements-docs.txt`. Also, this obviously means its `working-directory` should not be changed to docs/. 

aside: it's obviously a bit inefficient to install the iohub package just to install the docs-build deps, but I think it's okay unless/until the package deps become unwieldy or complex (which hopefully never happens).

Reply:
great catch, addressed.

regarding your aside: AFAIK, it is actually required to install your package to utilize autosummary features the way we use with sphinx. Otherwise importing package submodules fails in different doc source files.

Reply:
That does make sense, but I know that the copylot docs workflow does not install the package, just the build deps, and it works (unless its docs don't use the autosummary features? I know nothing about sphinx...)

Reply:
I mean, it's already in there, and it works, so I would vote for leaving it alone